### PR TITLE
Refactor/better stop in steprunner

### PR DESCRIPTION
**Pull request by rihter007**
Previous StepRunner had an issue with stop: if ones stops it and then tries to add a new target it will fail.
Also Stop used bulky sync.Once that had now possibility to understand if the StepRunner is stopped.

Fixes: https://github.com/linuxboot/contest/issues/69

### CODE REVIEW

File: pkg/runner/step_runner_test.go [hunk 1]
```diff
@@ -108,3 +113,189 @@ func (s *StepRunnerSuite) TestRunningStep() {
...
+		defer stepRunner.Stop()
+
+		stepRunner.Stop()
+		resultChan, err := stepRunner.Run(ctx,
+			s.NewStep(ctx, "test_step_label", stateFullStepName, nil),
+			emitter,
+			nil,
+			nil,
+		)
+		require.NoError(s.T(), err)
```
Comment (mimir-d) [inline pkg/runner/step_runner_test.go:297]: i dont have a strong opinion on this, but it sounds to me like it should be an error to try to run after stopped

**Review by mimir-d — COMMENTED** [review]
there's a couple questions to answer before approval

File: pkg/runner/step_runner_test.go [hunk 2]
```diff
@@ -108,3 +113,189 @@ func (s *StepRunnerSuite) TestRunningStep() {
...
+func (s *StepRunnerSuite) TestAddSameTargetMultipleTimes() {
+	ctx, cancel := xcontext.WithCancel(logrusctx.NewContext(logger.LevelDebug))
+	defer cancel()
+
+	const inputTargetID = "input_target_id"
+
+	err := s.RegisterStateFullStep(
+		func(ctx xcontext.Context, ch test.TestStepChannels, params test.TestStepParameters, ev testevent.Emitter, resumeState json.RawMessage) (json.RawMessage, error) {
+			_, err := teststeps.ForEachTarget(stateFullStepName, ctx, ch, func(ctx xcontext.Context, target *target.Target) error {
+				require.NotNil(s.T(), target)
```
[the remark this answers is not in the document]
Comment (rihter007) [reply pkg/runner/step_runner_test.go:126]: True, but I prefer require to be the same with non-suite tests

File: pkg/runner/step_runner_test.go (hunk 1 above)
Comment (rihter007) [reply pkg/runner/step_runner_test.go:297]: Maybe we should find a better name. Stop() means to close input channel or no more targets

**Pull request closed without merging**